audit/audit.cpp

```diff
@@ -147,7 +147,20 @@ audit::send_to_monitor(const std::string& stat)
                   % this->monitor_endpoint->address().to_string()
                   % this->monitor_endpoint->port();
 
-    this->socket->send_to(stat, *(this->monitor_endpoint));
+    this->socket->async_send_to(stat, *(this->monitor_endpoint), std::bind(&audit::handle_udp_send_callback,
+                                                                           shared_from_this(),
```

Comment:
We will need to wrap the callback in a strand.
There may be a chance that we will have a flurry of sends that may corrupt the data or trigger an asio internal assert if not protected.